cli/command/image/build_buildkit.go

```diff
@@ -0,0 +1,312 @@
+			if err != nil {
+				return err
+			}
+			defer response.Body.Close()
+			return nil
+		})
+	}
+
+	eg.Go(func() error {
+		defer func() { // make sure the Status ends cleanly on build errors
+			s.Close()
+		}()
+
+		buildOptions := imageBuildOptions(dockerCli, options)
+		buildOptions.Version = types.BuilderBuildKit
+		buildOptions.Dockerfile = dockerfileName
+		//buildOptions.AuthConfigs = authConfigs   // handled by session
+		buildOptions.RemoteContext = remote
+		buildOptions.SessionID = s.ID()
+		buildOptions.BuildID = buildID
+		return doBuild(ctx, eg, dockerCli, options, buildOptions)
+	})
+
+	return eg.Wait()
+}
+
+func doBuild(ctx context.Context, eg *errgroup.Group, dockerCli command.Cli, options buildOptions, buildOptions types.ImageBuildOptions) (finalErr error) {
+	response, err := dockerCli.Client().ImageBuild(context.Background(), nil, buildOptions)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	done := make(chan struct{})
+	defer close(done)
+	eg.Go(func() error {
+		select {
+		case <-ctx.Done():
+			return dockerCli.Client().BuildCancel(context.TODO(), buildOptions.BuildID)
+		case <-done:
+		}
+		return nil
+	})
+
+	t := newTracer()
+	ssArr := []*client.SolveStatus{}
+
+	displayStatus := func(displayCh chan *client.SolveStatus) {
+		var c console.Console
+		out := os.Stderr
+		// TODO: Handle interactive output in non-interactive environment.
+		consoleOpt := options.console.Value()
+		if cons, err := console.ConsoleFromFile(out); err == nil && (consoleOpt == nil || *consoleOpt) {
+			c = cons
+		}
+		// not using shared context to not disrupt display but let is finish reporting errors
+		eg.Go(func() error {
+			return progressui.DisplaySolveStatus(context.TODO(), c, out, displayCh)
+		})
+	}
+
+	if options.quiet {
+		eg.Go(func() error {
+			// TODO: make sure t.displayCh closes
+			for ss := range t.displayCh {
+				ssArr = append(ssArr, ss)
+			}
+			<-done
+			// TODO: verify that finalErr is indeed set when error occurs
+			if finalErr != nil {
+				displayCh := make(chan *client.SolveStatus)
+				go func() {
+					for _, ss := range ssArr {
+						displayCh <- ss
+					}
+					close(displayCh)
+				}()
+				displayStatus(displayCh)
+			}
+			return nil
+		})
+	} else {
+		displayStatus(t.displayCh)
+	}
+	defer close(t.displayCh)
+	err = jsonmessage.DisplayJSONMessagesStream(response.Body, os.Stdout, dockerCli.Out().FD(), dockerCli.Out().IsTerminal(), t.write)
+	if err != nil {
+		if jerr, ok := err.(*jsonmessage.JSONError); ok {
+			// If no error code is set, default to 1
+			if jerr.Code == 0 {
+				jerr.Code = 1
+			}
+			// if options.quiet {
+			// 	fmt.Fprintf(dockerCli.Err(), "%s%s", progBuff, buildBuff)
+			// }
+			return cli.StatusError{Status: jerr.Message, StatusCode: jerr.Code}
+		}
+		return err
+	}
+
+	return nil
+}
+
+func resetUIDAndGID(s *fsutil.Stat) bool {
+	s.Uid = uint32(0)
+	s.Gid = uint32(0)
+	return true
```

Comment:
was wondering why the boolean return was here, but guess this is to satisfy an interface? (reading from my phone so couldn't check easily)